diff --git a/drivers/soc/renesas/rzfive/cache.c b/drivers/soc/renesas/rzfive/cache.c
index 58ea0517edfc..e3701eebf923 100644
--- a/drivers/soc/renesas/rzfive/cache.c
+++ b/drivers/soc/renesas/rzfive/cache.c
@@ -69,7 +69,7 @@ static struct rzf_pma_plat pma_area[3];
 
 DEFINE_PER_CPU(struct andesv5_cache_info, cpu_cache_info) = {
 	.init_done = 0,
-	.dcache_line_size = SZ_32
+	.dcache_line_size = SZ_64
 };
 
 static uint32_t cpu_get_mcache_ctl_status(void)
